src/librustc_typeck/diagnostics.rs

```diff
@@ -3426,6 +3426,50 @@ parameters. You can read more about it in the API documentation:
 https://doc.rust-lang.org/std/marker/struct.PhantomData.html
 "##,
 
+E0436: r##"
+A struct can include `..` to indicate that you want to use a copy of some other
+struct for some of the values. This functional update syntax can only be used
+on struct types. It is an error to use it on other types, such as enums.
+
+Example of erroneous code:
+
+```compile_fail
+enum Foo {
+    A { x: u32, y: u32 }
```

Comment:
@GuillaumeGomez I'm sorry, I'm not quite seeing it. It looks to me like it's formatted the same way as in the [Enum docs](https://doc.rust-lang.org/book/enums.html). Could you explain what's wrong?